protocol/x/clob/keeper/equity_tier_limit.go

```diff
@@ -53,102 +53,127 @@ func (k *Keeper) InitializeEquityTierLimit(
+// tier limit on how many short term open orders a subaccount can have. Short-term fill-or-kill and immediate-or-cancel
+// orders never rest on the book and will always be allowed as they do not apply to the number of open orders that
+// equity tier limits enforce.
+func (k Keeper) ValidateSubaccountEquityTierLimitForShortTermOrder(ctx sdk.Context, order types.Order) error {
+	if order.RequiresImmediateExecution() {
+		return nil
+	}
+
+	equityTierLimits := k.GetEquityTierLimitConfiguration(ctx).ShortTermOrderEquityTiers
+	if len(equityTierLimits) == 0 {
+		return nil
+	}
+
+	equityTierLimit, err := k.getEquityTierLimit(ctx, order.GetSubaccountId(), equityTierLimits)
+	if err != nil {
+		return err
+	}
+
+	// For short term orders we just count how many orders exist on the memclob.
+	equityTierCount := k.MemClob.CountSubaccountShortTermOrders(ctx, order.GetSubaccountId())
+
+	// Verify that opening this order would not exceed the maximum amount of orders for the equity tier.
+	if lib.MustConvertIntegerToUint32(equityTierCount) >= equityTierLimit.Limit {
 		return errorsmod.Wrapf(
 			types.ErrOrderWouldExceedMaxOpenOrdersEquityTierLimit,
-			"Opening order would exceed equity tier limit of %d. Order id: %+v",
+			"Opening order would exceed equity tier limit of %d. Order count: %d, order id: %+v",
 			equityTierLimit.Limit,
+			equityTierCount,
 			order.GetOrderId(),
 		)
 	}
+	return nil
+}
 
-	equityTierCount := uint32(0)
-	if order.IsStatefulOrder() {
-		// For stateful orders we get the stateful order count.
-		// If this is `CheckTx` then we must also add the number of uncommitted stateful orders that this validator
-		// is aware of (orders that are part of the mempool but have yet to proposed in a block).
-		equityTierCount = k.GetStatefulOrderCount(ctx, order.OrderId.SubaccountId)
-		if !lib.IsDeliverTxMode(ctx) {
-			equityTierCountMaybeNegative := k.GetUncommittedStatefulOrderCount(ctx, order.OrderId) + int32(equityTierCount)
-			if equityTierCountMaybeNegative < 0 {
-				panic(
-					fmt.Errorf(
-						"Expected ValidateSubaccountEquityTierLimitForNewOrder for new order %+v to be >= 0. "+
-							"equityTierCount %d, statefulOrderCount %d, uncommittedStatefulOrderCount %d.",
-						order,
-						equityTierCountMaybeNegative,
-						k.GetStatefulOrderCount(ctx, order.OrderId.SubaccountId),
-						k.GetUncommittedStatefulOrderCount(ctx, order.OrderId),
-					),
-				)
-			}
-			equityTierCount = uint32(equityTierCountMaybeNegative)
-		}
+// ValidateSubaccountEquityTierLimitForStatefulOrder returns an error if adding the order would exceed the equity
+// tier limit on how many open orders a subaccount can have
+//
+// Note that the method is dependent on whether we are executing on `checkState` or on `deliverState` for
+// stateful orders. For `deliverState`, we sum:
+//   - the number of long term orders.
+//   - the number of triggered conditional orders.
```

Comment:
do we need to split out triggered and untriggered conditional orders?
(Is there a third class of conditional order that we are specifically not counting?)

Reply:
not that I know of. This was the present comment on the function doc for stateful orders